### DISCUSSION TITLE
serverSide getSession + Vercel deployment

### DISCUSSION THREAD

**Discussion by maximevast** [nextauthjs/next-auth#4199] [Help]
Hello !

I'm working with:

```json
"@vercel/next": "2.7.4",
"next-auth": "4.1.2",
```

My project currently handles authentication with `next-auth` `useSession` hook flawlessly, both in local and on the Vercel deployment.

It is worth mentioning that I don't have the `NEXTAUTH_URL` env variable set nor the `Automatically expose System Environment Variables` option checked, but I manually expose a `VERCEL: true` variable.
I do however have the `NEXTAUTH_URL` set locally.

Today, I tried to add some server side logic as follow:

```typescript
export async function getServerSideProps(context: NextPageContext) {
  const {req} = context
  const session = await getSession({req})

  if (session) {
    return {
      redirect: {destination: '/'},
    }
  }

  return {
    props: {},
  }
}
```

It is working locally, but not on Vercel. Here's the error message:

`message: 'request to http://localhost:3000/api/auth/session failed, reason: connect ECONNREFUSED 127.0.0.1:3000'`

What kind of configuration am I missing ? 

Best

**Comment by dubbajubba** [comment]
I don't think you need to destructure context when using `getSession()` in `getServerSideProps`. Have you tried to just pass context directly to `getSession()`?

```TSX
export async function getServerSideProps(context: NextPageContext) {
  const session = await getSession(context)

  if (session) {
    return {
      redirect: {destination: '/'},
    }
  }

  return {
    props: {},
  }
}
```

**Reply by maximevast** [reply]
I changed it but it did not fix the issue.

**Reply by dubbajubba** [reply]
Ok, sorry, was just browsing the discussions and saw your thread, noticed that part was different from mine.

Are you persisting your data to a database, or are you using the JWT strategy for handling authentication?

**Reply by dubbajubba** [reply]
Have you considered using the next-auth middleware? Not sure if it will work for your case, but I just removed getSession() from my page in favor of the middleware and it works very nicely.

https://next-auth.js.org/configuration/nextjs#middleware

**Reply by maximevast** [reply]
I don't think that's related to my configuration issue. THanks for the tip though

**Comment by maximevast** [comment]
Is it normal that the lambda attempts to reach localhost ? or should it use an external url to get the session?

**Reply by dubbajubba** [reply]
Yeah, I believe that is expected when running your app locally.

**Reply by maximevast** [reply]
No but I meant on Vercel

**Comment by breath-imager** [comment]
did you ever figure this out? I suspect I'm having the same issue and it's driving me mad
